source/MaterialXCore/Look.cpp

```diff
@@ -147,6 +146,17 @@ NodePtr MaterialAssign::getReferencedMaterial() const
     return resolveRootNameReference<Node>(getMaterial());
 }
 
+vector<OutputPtr> MaterialAssign::getMaterialOutputs() const
+{
+    vector<OutputPtr> materialOutputs;
+    NodeGraphPtr materialGraph = resolveRootNameReference<NodeGraph>(getMaterial());
+    if (materialGraph)
+    {
+        return materialGraph->getMaterialOutputs();
+    }
+    return materialOutputs;
```

Comment:
Note really a merge problem, but.. this method looks a bit strange, always returning an empty output vector if the referenced material is not a nodegraph? Is this to be worked on more when we always have explicit ouputs on all nodes? Maybe a comment to document that would be good in that case.

Reply:
Yes this is the case. Basically the new methods only handle nodegraphs and avoid nodes -- even if they have material outputs. If / when we move to having explicit outputs all the time this will need to be updated. I'll check the documentation.